### DISCUSSION TITLE
Replacing file and directory names

### DISCUSSION THREAD

**Discussion by sekerez** [your-tools/ruplacer#86] [Ideas]
I've been using this command for a while now and I love it.

Would you consider adding features to rename files and directories recursively in the same style?

I'm having trouble renaming multiple files/directories in a project according to a regex. Current commands (e.g. `rename`) are insufficient, as they don't have support for recursive operations.

**Comment by dmerejkowsky** [comment]
> Would you consider adding features to rename files and directories recursively in the same style?

Not really. Complex filesystem operations like these are very easy to get wrong and I think this is out of scope for ruplacer.

You may find `vidir` (from the `moreutils`) package useful.

**Comment by cgestes** [comment]
Or in zsh: ```zmv``` 😁

**Comment by wakuflair** [comment]
I have this need as well, so I try to add an additional option `--replace-path`.

You can try my fork:

https://github.com/wakuflair/ruplacer/tree/replace-path
